modules/ownadxBidAdapter.js

```diff
@@ -0,0 +1,102 @@
+import { parseSizesInput, isEmpty } from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js'
+
+/**
+ * @typedef {import('../src/adapters/bidderFactory.js').BidRequest} BidRequest
+ * @typedef {import('../src/adapters/bidderFactory.js').Bid} Bid
+ * @typedef {import('../src/adapters/bidderFactory.js').ServerResponse} ServerResponse
+ */
+
+const BIDDER_CODE = 'ownadx';
+const CUR = 'USD';
+const CREATIVE_TTL = 300;
+
+let tkn;
+let seatid;
+let sspid;
+let mtype = 0;
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    if (bid.mediaTypes[BANNER]) {
+      mtype = 1;
```

Comment:
This is an odd way to use the isBidRequestValid function.
Its purpose is to validate the bid request, it should not be to store param to use them in the buildRequests function.
The params are still available in the buildRequests function, why not use them directly in your validBidRequests.map function without storing them ?

The method seems to avoid the possibility to have bid requests with different seatId, tokenId or sspId
If only the last bid is invalid because it would not have all params, you would store incorrect param. Also, this would only support the last mediaTypes. 

Is there a reason I am missing ?

Reply:
@JulieLorin As per your suggestion we have made the changes.

Reply:
Thank you, this is much better that way